packages/common-all/src/types/workspace.ts

```diff
@@ -130,6 +130,7 @@ export enum LookupSelectionType {
 
 export type NoteLookupConfig = {
   selectionType: LookupSelectionType;
+  leaveTrace?: boolean;
```

Comment:
heads up that there will be significant changes to the configurations and we will be introducing a lot of changes to it incrementally for the next few weeks.

I have written a config convention guide here: https://github.com/dendronhq/dendron-site/pull/196/files

For boolean configs in particular, we want to make it a required property of the config namespace so that defaults are explicit in the configuration and we don't have to take account for assumed defaults in the implementation.
This helps the next person down the line who has to read the code have a clear understanding of what behavior is expected.

On naming, we want to try limiting our selection of verbs in the transitive verb form names of configs, but in this case it's a bit awkward to name it `enableLeaveTrace` or something... so I think this is fine for now. Welcome to suggest anything better if you can come up with it after going through the convention guide 🙂

Reply:
Agreed. 
Although what if we provide a default value in config file without making it a required? The reason behind this is if this config is not present in the dendron.yml, it should not throw an error to user on initialization. 
Somewhat similar to the `lookupConfirmVaultOnCreate`. Even though the default value is present on init, it is not a required config. What do you think?

Reply:
My reasoning for having all configurations that have bounded choices is because it is clear what it does. This is true for configs that are enums as well.

One anecdotal case is `selectionType` of lookup. This used to be either `selectionExtract` or `selection2link`. As a user that has not seen how this is implemented, and without reading the documentation, what do you think omitting this setting will do?

Putting myself in a moderately engaged user's standpoint I still would think omitting this setting will mean the selections will not affect how lookup is done. That was not the case until I added a `none` case. Even for an experience user and a developer, I often feel like I don't know what a config is doing unless I've been using that a lot or have worked on the area related to it.

Having documentation that explicitly states what it does is great and we strive to do so, but even with a best intent we sometimes forget about documentation or wrongly assume how things work. If we don't have control over what the user will actually do here (decided to read the docs), we should have safety measures. and my argument is being explicit and letting the user know something is wrong on init is a better way of handling this.

Your reasoning is valid, but I think it complicates the behavior in a way that it's hard to expect what's going to happen if you don't have prior knowledge of how Dendron is working internally. I would lean towards properly handling errors on init if a required property is missing (this right now is done by validating the json schema, but this is something a user can choose not to do). _Providing accurate and helpful errors when things go wrong is a better UX than allowing loosely defined behavior and keeping the user happy (but oblivious)_ is my argument here.

Another concern is consistency. If we allow one config to be optional for the sake of preventing a potential error, we are opening the gates to all the configs that have bounded choices to be optional. Of those, some will consequently not be optional sometimes. the expectations don't match for all config keys. This is a problem for the user but also a problem for us (and all future contributors) developing and adding new configs.

`lookupConfirmVaultOnCreate` will be required moving forward, btw (in fact, almost all booleans will be: https://github.com/dendronhq/dendron/pull/1295#issuecomment-919140938)
![image](https://user-images.githubusercontent.com/1219789/134753114-2aa101fe-6206-4daa-ad9b-cc708624d309.png)

Reply:
Very well thought. Thanks :)